rpcs3/Emu/Cell/Modules/cellSearch.cpp

```diff
@@ -223,11 +230,135 @@ error_code cellSearchStartListSearch(CellSearchListSearchType type, CellSearchSo
 
 	const u32 id = *outSearchId = idm::make<search_object_t>();
 
-	sysutil_register_cb([=](ppu_thread& ppu) -> s32
+	sysutil_register_cb([=, content_map = g_fxo->get<ContentIdMap>()](ppu_thread& ppu) -> s32
 	{
+		auto curr_search = idm::get<search_object_t>(id);
 		vm::var<CellSearchResultParam> resultParam;
-		resultParam->searchId = id;
-		resultParam->resultNum = 0; // TODO
+		resultParam->searchId  = id;
+		resultParam->resultNum = 0; // Set again later
+
+		std::function<void(const std::string&)> searchInFolder = [&, type](const std::string& vpath)
+		{
+			// TODO: this is just a workaround. On a real PS3 the playlists seem to be stored in dev_hdd0/mms/db/metadata_db_hdd
+			for (auto&& item : fs::dir(vfs::get(vpath)))
+			{
+				item.name = vfs::unescape(item.name);
+
+				if (item.name == "." || item.name == "..")
+				{
+					continue;
+				}
+
+				if (!item.is_directory)
+				{
+					continue;
+				}
+
+				// TODO - Identify sorting method and insert the appropriate values where applicable
+				const std::string item_path(vpath + "/" + item.name);
+
+				// Count files
+				u32 numOfItems = 0;
+				for (auto&& file : fs::dir(vfs::get(item_path)))
+				{
+					file.name = vfs::unescape(file.name);
+
+					if (file.name == "." || file.name == ".." || file.is_directory)
+					{
+						continue;
+					}
+
+					numOfItems++;
+				}
+
+				const u64 hash = std::hash<std::string>()(item_path);
+				auto found = content_map->find(hash);
+				if (found == content_map->end()) // content isn't yet being tracked
+				{
+					auto ext_offset = item.name.find_last_of('.'); // used later if no "Title" found
```

Comment:
I think, this line can be placed closer to usage, to not search for the dot if unused